bin/run-metadetect-on-slices

```diff
@@ -19,6 +21,77 @@ from ngmix.medsreaders import MultiBandNGMixMEDS, NGMixMEDS
 from pizza_cutter.files import StagedOutFile, makedir_fromfile
 
 
+def _process(*,
+             meds_files,
+             config,
+             output_path,
+             seed,
+             part,
+             log_level,
+             meds_range,
+             tmpdir):
+
+    with open(config, 'r') as fp:
+        _config = yaml.load(fp.read(), Loader=yaml.Loader)
+
+    logging.basicConfig(
+        stream=sys.stdout, level=getattr(logging, log_level.upper()))
+
+    meds_list = []
+    for f in meds_files:
+
+        if tmpdir is not None:
+            usename = os.path.join(tmpdir, os.path.basename(f))
+            print('copying %s -> %s' % (f, usename))
+            shutil.copy(f, tmpdir)
```

Comment:
Should we be using the StagedInFile class you have lying around?

Reply:
Good idea, but we wouldn't be able to use it in a context manager since there are N files to be staged int, so it would not be so clean. Might as well do it this way and let the TemporaryDirectory() do the cleanup

Reply:
actually, it would make the code more clean because it works the same with for tmpdir=None or not

Reply:
Right. Seems like a lot of work for a single if statement. I am ok as is.

Reply:
not sure how to make it work though, the files go out of scope...

Reply:
oh, duh, I had a `__del__` defined in there, just need to remove it so it only does cleanup if used as a context manager